blockprotocol/blockprotocol
# Add internal API client to the BP site

## 🌟 What is the purpose of this PR?




This PR adds an internal API client generator, and the generated internal API client to the `@local` directory, so that the BP site can use it to make requests to the internal API.

## 🔗 Related links





- [Connect BP Site to internal API client](https://app.asana.com/0/1203543021352032/1203672409051076/f) _(internal)_
- [Setup "billing" feature flag in the BP site](https://app.asana.com/0/1203543021352032/1203655456562769/f) _(internal)_

## 🔍 What does this change?




See commits.

## 📜 Does this require a change to the docs?



Not that I'm aware of.


## 🐾 Next steps



- [Implement MVP billing page in BP site](https://app.asana.com/0/1203543021352032/1203655456562773)

## 🛡 What tests cover this?



The functionality this PR enables was manually tested.

## ❓ How to test this?



1. Run the internal API developer environment locally (see the repo README for instructions)
2. Run the BP site developer environment locally
3. Ensure you've added the required `.env` variables in your `.env.local` as specified in the README. This should include values for `INTERNAL_API_BASE_URL`, `INTERNAL_API_KEY` and `NEXT_PUBLIC_BILLING_FEATURE_FLAG`.
5. Use an HTTP client to send requests to `localhost:3000/api/internal/openapi.json` or some other internal API route, and ensure you receive a response from the internal API and not the Next.js middleware

## 📹 Demo




https://user-images.githubusercontent.com/42802102/212307643-1e525b7f-ec83-45b3-9ff3-2611c8480a82.mov
